Source/UI/VirtusizeInPageStandard.swift

```diff
@@ -299,26 +344,111 @@ public class VirtusizeInPageStandard: VirtusizeInPageView {
         setLoadingScreen(loading: true)
     }
 
-    public override func setInPageText() {
-        guard let storeProduct = Virtusize.storeProduct,
-              let i18nLocalization = Virtusize.i18nLocalization else {
-            self.showErrorScreen()
-            return
-        }
-
-        self.productImageView.setImage(storeProduct: storeProduct, localImageUrl: Virtusize.product?.imageURL) {
-            self.setLoadingScreen(loading: false)
-            self.setMessageLabelTexts(
-				storeProduct,
-				i18nLocalization,
-				Virtusize.sizeComparisonRecommendedSize,
-				Virtusize.bodyProfileRecommendedSize?.sizeName
-			)
-        }
-    }
-
+	public override func setInPageText() {
```

Comment:
That part is really complicated and hard to read.
Can you break it down to avoid the nested conditional hell?

Reply:
I broke it down into smaller functions and added some comments. Hope it will make sense now.